ntoskrnl/mm/ARM3/procsup.c

```diff
@@ -890,6 +901,15 @@ MiInsertSharedUserPageVad(VOID)
         return Status;
     }
 
+    Status = PsChargeProcessNonPagedPoolQuota(Process, sizeof(MMVAD_LONG));
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("Ran out of quota.\n");
```

Comment:
You can't fail here, after the VAD was already inserted!
Also, didn't that cause a problem with PsInitialSystemProcess not being set up? How was that fixed?